payments/schema/mutations.py

```diff
@@ -416,6 +416,28 @@ def mutate_and_get_payload(cls, root, info, **input):
         return ConfirmPaymentMutation(url=payment_url)
 
 
+class CancelOrderMutation(graphene.ClientIDMutation):
+    class Input:
+        order_number = graphene.String(required=True)
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **input):
+        order_number = input.get("order_number", None)
+        try:
+            order = Order.objects.get(order_number=order_number)
```

Comment:
The order numbers are sufficiently random hopefully, so that you can't just guess random orders to cancel?

Reply:
[Should be](https://github.com/City-of-Helsinki/berth-reservations/blob/develop/payments/utils.py#L183-L186)
```python
def generate_order_number() -> str:
    t = time.time() * 1000000 * random.uniform(1, 1000)
    b = base64.b32encode(struct.pack(">Q", int(t)).lstrip(b"\x00")).strip(b"=").lower()
    return b.decode("utf8")
```